src/shogun/lib/any.h

```diff
@@ -71,6 +71,54 @@ namespace shogun
 		NON_OWNING
 	};
 
+	namespace any_detail
+	{
+
+		struct by_default
+		{
+		};
+
+		struct general : by_default
+		{
+		};
+
+		struct more_important : general
+		{
+		};
+
+		struct maybe_most_important : more_important
+		{
+		};
+
+		template <class T>
+		auto compare_impl(by_default, T& lhs, T& rhs) = delete;
```

Comment:
`auto` return is only c++14. but no worries just up the required version here:
https://github.com/shogun-toolbox/shogun/blob/develop/CMakeLists.txt#L35

Reply:
Oh wow so this happened? :dancing_men: